apps/system/js/audio_channel_policy.js

```diff
@@ -0,0 +1,273 @@
+     *
+     * @param {AudioChannelController} audioChannel
+     * A new channel you want to handel.
+     * @return {Object} The policies for active and new audio channels.
+     * The object is like below
+     * {
+     *   activeAudioChannels: [
+     *     { isAllowedToPlay: false,
+     *       isResumedWhenOtherEnds: true,
+     *       audioChannel: fm
+     *     },
+     *     {
+     *       isAllowedToPlay: true,
+     *       audioChannel: notification 
+     *     }
+     *   ],
+     *   newAudioChannel: {
+     *     isAllowedToPlay: true,
+     *     isNeededToFadeOut: true,
+     *     audioChannel: music 
+     *   }
+     * }
+     */
+    compete: function(newAudioChannel, activeAudioChannels, options) {
+      var policies = { newAudioChannel: null, activeAudioChannels: null };
+      var newAudioChannelName = newAudioChannel.channelName;
+      var isAllowedToPlayForNewAduioChannel = true;
+      var isNeededToFadeOutForNewAduioChannel = false;
+      if (activeAudioChannels.size) {
+        policies.activeAudioChannels = [];
+        activeAudioChannels.forEach((activeAudioChannel) => {
+          var activeAudioChannelName = activeAudioChannel.channelName;
+          var results = AUDIO_CHANNEL_COMPETITION_RESULES
+            [activeAudioChannelName][newAudioChannelName];
+          var isNeededToFadeOut = this._isNeededToFadeOutForActiveAudioChannel(
+              activeAudioChannelName, newAudioChannelName);
+          var isNeededToVibrate = this._isNeededToVibrateForActiveAudioChannel(
+              activeAudioChannelName, newAudioChannelName);
+          var isResumedWhenOtherEnds;
+          if (!results.activeAudioChannel) {
+            isResumedWhenOtherEnds =
+              this._isNeededToResumeWhenOtherEndsForActiveAudioChannel(
+                activeAudioChannelName,newAudioChannelName
+              );
+          }
+          policies.activeAudioChannels.push({
+            isAllowedToPlay: results.activeAudioChannel,
+            isNeededToFadeOut: isNeededToFadeOut,
+            isNeededToVibrate: isNeededToVibrate,
+            isResumedWhenOtherEnds: isResumedWhenOtherEnds,
+            audioChannel: activeAudioChannel
+          });
+          // Deconflict the conflict policies for the new audio channel.
+          // For example, we have a new audio channel `normal` and
+          // two active audio channels `content` and `notification`.
+          // `AUDIO_CHANNEL_COMPETITION_RESULES.content.normal` is `false`,
+          // but `AUDIO_CHANNEL_COMPETITION_RESULES.notification.normal`
+          // is `true`.
+          // And we get conflict policies for the `normal` audio channel.
+          // Once we get conflcts, we will take higher priority policy.
+          // `PAUSE` is higher than `PLAY`,
+          // and doing fade out is higher than not doing fade out.
+          isAllowedToPlayForNewAduioChannel =
+            isAllowedToPlayForNewAduioChannel && results.newAudioChannel;
```

Comment:
isAllowedToPlayForNewAduioChannel is true by your definition, why do we need to && here?